gravitee-node-monitoring/src/test/java/io/gravitee/node/monitoring/DefaultProbeRegistryTest.java

```diff
@@ -0,0 +1,219 @@
+package io.gravitee.node.monitoring;
+
+import static io.gravitee.node.monitoring.DefaultProbeRegistry.SAFEGUARD_DELAY;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+
+import io.gravitee.node.api.healthcheck.Probe;
+import io.gravitee.node.api.healthcheck.ProbeManager;
+import io.gravitee.node.api.healthcheck.Result;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * @author Jeoffrey HAEYAERT (jeoffrey.haeyaert at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@ExtendWith(MockitoExtension.class)
+class DefaultProbeRegistryTest {
+
+    protected static final int CACHE_DURATION_MS = 30000;
+
+    @Mock
+    private ProbeManager probeManager;
+
+    private DefaultProbeRegistry cut;
+
+    @Test
+    void should_complete_and_check_all_probes_when_evaluate() {
+        final Probe probe1 = mock(Probe.class);
+        final Probe probe2 = mock(Probe.class);
+
+        when(probe1.check()).thenReturn(CompletableFuture.completedFuture(Result.healthy()));
+        when(probe2.check()).thenReturn(CompletableFuture.completedFuture(Result.healthy()));
+        when(probeManager.getProbes()).thenReturn(List.of(probe1, probe2));
+
+        cut = new DefaultProbeRegistry(0, probeManager);
+
+        final CompletableFuture<Map<Probe, Result>> result = cut.evaluate();
+
+        assertThat(result)
+            .isCompletedWithValueMatching(probeResultMap -> {
+                assertThat(probeResultMap.get(probe1)).isEqualTo(Result.healthy());
+                assertThat(probeResultMap.get(probe2)).isEqualTo(Result.healthy());
+                return true;
+            });
+
+        verify(probe1).check();
+        verify(probe2).check();
+    }
+
+    @Test
+    void should_complete_with_probe_unhealthy_when_probe_unhealthy() {
+        final Probe probe1 = mock(Probe.class);
+        final Probe probe2 = mock(Probe.class);
+
+        when(probe1.check()).thenReturn(CompletableFuture.completedFuture(Result.unhealthy("unhealthy")));
+        when(probe2.check()).thenReturn(CompletableFuture.completedFuture(Result.healthy()));
+        when(probeManager.getProbes()).thenReturn(List.of(probe1, probe2));
+
+        cut = new DefaultProbeRegistry(0, probeManager);
+
+        final CompletableFuture<Map<Probe, Result>> result = cut.evaluate();
+
+        assertThat(result)
```

Comment:
```
should_complete_and_check_all_probes_when_evaluate
should_complete_with_probe_unhealthy_when_probe_unhealthy
```

They are 2 different cases. Don't you feel that it would complexify the comprehension of what the test does?